src/initialization/InitDistribution.cpp

```diff
@@ -551,15 +592,36 @@ namespace impactx
             std::string unit_type;  // System of units
             pp_dist.get("units", unit_type);
 
+            // phase space distribution
             distribution::KnownDistributions dist = initialization::read_distribution(pp_dist);
             std::string distribution;
             pp_dist.get("distribution", distribution);
 
+            // spin distribution
+            amrex::ParticleReal polarization_x = 0.0_prt,
+                                polarization_y = 0.0_prt,
+                                polarization_z = 0.0_prt;
+            pp_dist.queryWithParser("polarization_x", polarization_x);
+            pp_dist.queryWithParser("polarization_y", polarization_y);
+            pp_dist.queryWithParser("polarization_z", polarization_z);
+            amrex::ParticleReal pmag = 0.0_prt;
+            amrex::ParticleReal kappa = 0.0_prt;
+
+            // magnitude of the polarization vector (= polarization magnitude)
+            pmag = std::sqrt(polarization_x*polarization_x+polarization_y*polarization_y+polarization_z*polarization_z);
+
+            if (pmag < 1) {
+               kappa = distribution::SpinvMF::inverse_Langevin(pmag);
+            } else {
+                throw std::runtime_error("Initial polarization magnitude must currently be < 1.");
+            }
+            distribution::SpinvMF spin_dist(polarization_x, polarization_y, polarization_z, kappa);
+
             amrex::Long npart = 0;  // Number of simulation particles
             if (distribution != "empty")
             {
                 pp_dist.getWithParser("npart", npart);
-                add_particles(bunch_charge, dist, npart);
+                add_particles(bunch_charge, dist, npart, spin_dist);
```

Comment:
How do we know there is no `spin_dist` at all? `pmag` is exactly zero?

In that case, we use:
```suggestion
                add_particles(bunch_charge, dist, npart);
```